src/models/role.js

```diff
@@ -0,0 +1,142 @@
+import router from 'umi/router';
+import { message } from 'antd';
+import { query as queryRoles, addRole, queryById, updateRole, deleteById, menuAuth, menuAuthGet, permissionAuth, permissionAuthGet } from '@/services/role';
+export default {
+  namespace: 'role',
+  state: {
+    data: {
+      list: [],
+      pagination: {},
+    },
+    role: {},
+    roleMenuAuths: [],
+    rolePermissAuths: [],
+  },
+
+  effects: {
+    *fetch({ payload }, { call, put }) {
+      const response = yield call(queryRoles, payload);
+      yield put({
+        type: 'fetchSave',
+        payload: response,
+      });
+    },
+    *add({ payload }, { call, put }) {
+      const response = yield call(addRole, payload);
+      console.log(response);
+      if (response.code === 0) {
+        message.success('保存成功', 2);
+        router.push('/permission-management/role/list');
+      } else {
+        message.error(response.msg);
+      }
+    },
+    *get({ payload }, { call, put }) {
+      const response = yield call(queryById, payload);
+      yield put({
+        type: 'getSave',
+        payload: response,
+      });
+    },
+    *update({ payload }, { call, put }) {
```

Comment:
Codacy found an issue: ['put' is defined but never used.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=6006324)